File: benchmarks/sieve/sieve.py
def sieve(n):
    is_prime = [True] * (n + 1)
    is_prime[0] = False
    is_prime[1] = False

    i = 2
    while i * i <= n:
        if is_prime[i]:
            for j in range(i * i, n + 1, i):
                is_prime[j] = False
        i += 1

    count = 0
    for i in range(2, n + 1):
        if is_prime[i]:
            count += 1
    return count

n = 6000000
count = sieve(n)
print(f"primes up to {n}: {count}")

File: benchmarks/sieve/sieve.go
package main

import "fmt"

func sieve(n int) int {
	isPrime := make([]bool, n+1)
	for i := range isPrime {
		isPrime[i] = true
	}
	isPrime[0] = false
	isPrime[1] = false

	for i := 2; i*i <= n; i++ {
		if isPrime[i] {
			for j := i * i; j <= n; j += i {
				isPrime[j] = false
			}
		}
	}

	count := 0
	for i := 2; i <= n; i++ {
		if isPrime[i] {
			count++
		}
	}
	return count
}

func main() {
	n := 6000000
	count := sieve(n)
	fmt.Printf("primes up to %d: %d\n", n, count)
}
